# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/url-classifier/UrlClassifierFeatureTrackingAnnotation.cpp
# Commit: f2df73fda468
# Full Hash: f2df73fda46859150605f33bcdaeb49ab7037720
# Author: Jean-Yves Avenard <jyavenard@mozilla.com>
# Date: 2019-10-19 21:40:34
# Regressor Bug: 1588899
# File Overlap Count: 1
# Description:
#   Bug 1588899 - P1. Move classification flags related method to nsIClassifiedChannel. r=Ehsan,baku
#   
#   This is where it should have been in the first place. Those attributes belong there.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D49577
# ==============================================================================

diff -r b98968eb71d3 -r f2df73fda468 netwerk/url-classifier/UrlClassifierFeatureTrackingAnnotation.cpp
--- a/netwerk/url-classifier/UrlClassifierFeatureTrackingAnnotation.cpp	Thu Oct 17 15:24:31 2019 +0000
+++ b/netwerk/url-classifier/UrlClassifierFeatureTrackingAnnotation.cpp	Sat Oct 19 04:30:24 2019 +0000
@@ -12,6 +12,7 @@
 #include "mozilla/StaticPrefs_privacy.h"
 #include "mozilla/StaticPtr.h"
 #include "mozilla/net/UrlClassifierCommon.h"
+#include "nsIClassifiedChannel.h"
 #include "nsContentUtils.h"
 
 namespace mozilla {
@@ -127,18 +128,21 @@
   static std::vector<UrlClassifierCommon::ClassificationData>
       sClassificationData = {
           {NS_LITERAL_CSTRING("ads-track-"),
-           nsIHttpChannel::ClassificationFlags::CLASSIFIED_TRACKING_AD},
+           nsIClassifiedChannel::ClassificationFlags::CLASSIFIED_TRACKING_AD},
           {NS_LITERAL_CSTRING("analytics-track-"),
-           nsIHttpChannel::ClassificationFlags::CLASSIFIED_TRACKING_ANALYTICS},
+           nsIClassifiedChannel::ClassificationFlags::
+               CLASSIFIED_TRACKING_ANALYTICS},
           {NS_LITERAL_CSTRING("social-track-"),
-           nsIHttpChannel::ClassificationFlags::CLASSIFIED_TRACKING_SOCIAL},
+           nsIClassifiedChannel::ClassificationFlags::
+               CLASSIFIED_TRACKING_SOCIAL},
           {NS_LITERAL_CSTRING("content-track-"),
-           nsIHttpChannel::ClassificationFlags::CLASSIFIED_TRACKING_CONTENT},
+           nsIClassifiedChannel::ClassificationFlags::
+               CLASSIFIED_TRACKING_CONTENT},
       };
 
   uint32_t flags = UrlClassifierCommon::TablesToClassificationFlags(
       aList, sClassificationData,
-      nsIHttpChannel::ClassificationFlags::CLASSIFIED_TRACKING);
+      nsIClassifiedChannel::ClassificationFlags::CLASSIFIED_TRACKING);
 
   UrlClassifierCommon::SetTrackingInfo(aChannel, aList, aHashes);
 